From 8cc7ada0161a7af38351d1e70516975402f3c5c5 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Fri, 26 Feb 2021 18:37:41 +0000
Subject: [PATCH] libdnf/config.h: include the correct header on musl

Problem reported at https://github.com/rpm-software-management/libdnf/issues/1146,
but this patch isn't quite the fix.

Upstream-Status: Inappropriate
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 libdnf/config.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/libdnf/config.h b/libdnf/config.h
index 16121f6f..51623ce4 100644
--- a/libdnf/config.h
+++ b/libdnf/config.h
@@ -18,7 +18,12 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
+#include <features.h>
+#ifdef __GLIBC__
 #include <bits/wordsize.h>
+#else
+#include <bits/reg.h>
+#endif
 
 #if __WORDSIZE == 32
 #include "config-32.h"
